react-web/src/components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { ComponentProps, forwardRef, useState } from "react";
+import TextAreaAutoSize from "@mui/material/TextareaAutosize";
+import classNames from "classnames";
+
+import Icons from "components/Icons/Icons";
+import ArrowTooltip from "components/Tooltip/Tooltip";
+import HelperText from "../HelperText/HelperText";
+
+import "./TextArea.scss";
+import { useFormContext } from "react-hook-form";
+
+export interface TextAreaProps extends ComponentProps<"textarea"> {
+  minRows?: number;
+  maxRows?: number;
+  tooltipText?: string;
+  showInfoIcon?: boolean;
+}
+
+const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
+  function TextArea(
+    {
+      name = "",
+      value,
+      onChange,
+      placeholder,
+      minRows = 2,
+      maxRows = 5,
+      className,
+      tooltipText = "",
+      required = false,
+      showInfoIcon = false,
+    },
+    ref
+  ) {
+    const {
+      formState: { errors },
+    } = useFormContext();
+
+    const [active, setActive] = useState(false);
+
+    return (
+      <div
+        className={classNames("text-area-wrapper relative", {
+          error: errors?.[name],
+          active: active,
+        })}
+        onClick={(_) => {
+          setActive(true);
+          document.getElementById(name || "")?.focus();
+        }}
```

Comment:
Text area needs to be focused on clicking anywhere within the text area component. Since this is a forwardRef component, state of the referred component cannot be updated using `ref`. Hence ignoring this suggestion now